src/rustc_stderr.rs

```diff
@@ -125,20 +122,16 @@ pub(crate) fn filter_annotations_from_rendered(rendered: &str) -> std::borrow::C
 }
 
 pub(crate) fn process(file: &Path, stderr: &[u8]) -> Diagnostics {
-    let stderr = std::str::from_utf8(stderr).unwrap();
-    let mut rendered = String::new();
+    let mut rendered = Vec::new();
     let mut messages = vec![];
     let mut messages_from_unknown_file_or_line = vec![];
-    for line in stderr.lines() {
-        if line.starts_with('{') {
-            match serde_json::from_str::<RustcMessage>(line) {
+    for (line_number, line) in stderr.split_inclusive(|&c| c == b'\n').enumerate() {
```

Comment:
`&[u8]` implements `BufRead` which has a `lines` method, can't we use that?

Reply:
because that produces `String`s again

Reply:
Ah, bummer.

This is not entirely identical though, `lines()` also has handling for `\r`.

Reply:
I switched to bstr's `lines()` function which works correctly